packages/flutter_tools/lib/src/resident_web_runner.dart

```diff
@@ -120,78 +129,229 @@ class ResidentWebRunner extends ResidentRunner {
       printError(message);
       return 1;
     }
-    // Start the web compiler and build the assets.
-    final bool success = await webCompilationProxy.initialize(
-      projectDirectory: flutterProject.directory,
-    );
-    if (!success) {
-      throwToolExit('Failed to compile for the web.');
-    }
-    _lastCompiled = DateTime.now();
-    final AssetBundle assetBundle = AssetBundleFactory.instance.createBundle();
-    final int build = await assetBundle.build();
-    if (build != 0) {
-      throwToolExit('Error: Failed to build asset bundle.');
-    }
-    await writeBundle(fs.directory(getAssetBuildDirectory()), assetBundle.entries);
-
-    // Step 2: Start an HTTP server
-    _server = WebAssetServer(flutterProject, target, ipv6);
-    await _server.initialize();
-
-    // Step 3: Spawn an instance of Chrome and direct it to the created server.
-    final String url = 'http://localhost:${_server.port}';
-    final Chrome chrome = await chromeLauncher.launch(url);
-    final ChromeTab chromeTab = await chrome.chromeConnection.getTab((ChromeTab chromeTab) {
-      return chromeTab.url.contains(url); // we don't care about trailing slashes or #
-    });
-    _connection = await chromeTab.connect();
-    _connection.onClose.listen((WipConnection connection) {
-      exit();
-    });
-
-    // We don't support the debugging proxy yet.
+    Status buildStatus;
+    try {
+      buildStatus = logger.startProgress('Building application for the web...', timeout: null);
+      _webFs = await webFsFactory(
+        target: target,
+        flutterProject: flutterProject,
+        buildInfo: debuggingOptions.buildInfo,
+      );
+      if (supportsServiceProtocol) {
+        _debugConnection = await _webFs.runAndDebug();
+      }
+    } catch (err, stackTrace) {
+      printError(err.toString());
+      printError(stackTrace.toString());
+      throwToolExit('Failed to build application for the web.');
+    } finally {
+      buildStatus.stop();
+    }
     appStartedCompleter?.complete();
     return attach(
       connectionInfoCompleter: connectionInfoCompleter,
       appStartedCompleter: appStartedCompleter,
     );
   }
 
+  @override
+  Future<int> attach({
+    Completer<DebugConnectionInfo> connectionInfoCompleter,
+    Completer<void> appStartedCompleter,
+  }) async {
+    // Cleanup old subscriptions
+    try {
+      await _debugConnection?.vmService?.streamCancel('Stdout');
+    } on vmservice.RPCError {
+      // Ignore errors here
+    }
+    try {
+      await _debugConnection?.vmService?.streamListen('Stdout');
+    } on vmservice.RPCError  {
+      // Ignore errors here
```

Comment:
Why is it safe to ignore errors?

Reply:
These throw if there isn't anything listening, which is fine because that is what we want to ensure. Will add a comment to the code